test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -54,101 +65,156 @@ public class PassFailJFrame {
+     * @param columns              Number of columns of the instructional
+     *                             JTextArea
+     * @param testTimeOutInMinutes timeout of the test where time is specified in
+     *                             minutes.
+     * @throws HeadlessException         HeadlessException
+     * @throws InterruptedException      exception thrown for invokeAndWait
+     * @throws InvocationTargetException exception thrown for invokeAndWait
      */
     public PassFailJFrame(String title, String instructions,
-                          int maxRowLength, int maxStringLength,
-                          int timeoutMinutes) throws HeadlessException,
+                          int rows, int columns,
+                          int testTimeOutInMinutes) throws HeadlessException,
             InterruptedException, InvocationTargetException {
-        this.maxRowLength = maxRowLength;
-        this.maxStringLength = maxStringLength;
-        this.timeoutMinutes = timeoutMinutes;
 
         if (isEventDispatchThread()) {
-            createUI(title, instructions, maxRowLength, maxStringLength, timeoutMinutes);
+            createUI(title, instructions, rows, columns, testTimeOutInMinutes);
         } else {
-            invokeAndWait(() -> {
-                createUI(title, instructions, maxRowLength, maxStringLength, timeoutMinutes);
-            });
+            invokeAndWait(() -> createUI(title, instructions, rows, columns,
+                    testTimeOutInMinutes));
         }
     }
 
     private static void createUI(String title, String instructions,
-                                 int maxRowLength, int maxStringLength,
-                                 int timeoutMinutes) {
+                                 int rows, int columns,
+                                 int timeoutInMinutes) {
         frame = new JFrame(title);
         frame.setLayout(new BorderLayout());
-        instructionsText = new JTextArea(instructions, maxRowLength, maxStringLength);
+        JTextArea instructionsText = new JTextArea(instructions, rows, columns);
         instructionsText.setEditable(false);
-        instructionsText.setFocusable(false);
-        frame.add(instructionsText, BorderLayout.NORTH);
+        instructionsText.setLineWrap(true);
+
+        int testTimeout = (int) TimeUnit.MINUTES.toMillis(timeoutInMinutes);
+
+        final JLabel testTimeoutLabel = new JLabel(String.format("Test " +
+                "timeout: %s", convertMillisToTimeStr(testTimeout)), JLabel.CENTER);
+        final long startTime = System.currentTimeMillis();
+        timer.setDelay(1000);
+        timer.addActionListener((e) -> {
+            int leftTime = testTimeout - (int) (System.currentTimeMillis() - startTime);
+            if ((leftTime < 0) || failed) {
+                timer.stop();
+                testFailedReason = "Failure Reason:\n Timeout " +
+                        "User did not perform testing.";
```

Comment:
A suggestion: break string at the line break.
```suggestion
                testFailedReason = "Failure Reason:\n"
                        + "Timeout - User did not perform testing.";
```